src/main/java/com/ericsson/ei/rules/RuleTestService.java

```diff
@@ -0,0 +1,114 @@
+package com.ericsson.ei.rules;
+
+import com.ericsson.ei.exception.InvalidRulesException;
+import com.ericsson.ei.handlers.EventHandler;
+import com.ericsson.ei.handlers.EventToObjectMapHandler;
+import com.ericsson.ei.jmespath.JmesPathInterface;
+import com.ericsson.ei.queryservice.ProcessAggregatedObject;
+import org.json.JSONArray;
+import org.json.JSONException;
+import org.json.JSONObject;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.stereotype.Component;
+
+import java.io.IOException;
+import java.util.ArrayList;
+import java.util.List;
+
+@Component
+public class RuleTestService implements IRuleTestService {
+    private static final Logger LOGGER = LoggerFactory.getLogger(RuleTestService.class);
+
+    @Autowired
+    private JmesPathInterface jmesPathInterface;
+
+    @Autowired
+    private EventHandler eventHandler;
+
+    @Autowired
+    private ProcessAggregatedObject processAggregatedObject;
+
+    @Autowired
+    private EventToObjectMapHandler eventToObjectMapHandler;
+
+    @Override
+    public String prepareAggregatedObject(JSONArray listRulesJson, JSONArray listEventsJson)
+            throws JSONException, IOException, InvalidRulesException {
+        eventHandler.getRulesHandler().setParsedJson(listRulesJson.toString());
+        String response = "";
+
+        String templateName = validateRuleTemplateNames(listRulesJson);
+
+        prepareEventsForTestAggregation(listEventsJson, templateName);
+
+        List<String> responseList = processAggregatedObject.getAggregatedObjectByTemplateName(templateName);
+        response = responseList.toString();
+
+        // Delete the aggregated object
+        processAggregatedObject.deleteAggregatedObject(templateName);
+        // Delete the event object mapper
+        eventToObjectMapHandler.deleteEventObjectMap(templateName);
+
+        return response;
+    }
+
+    /**
+     * This method iterates through the list of rules to extract the template name from each rule.
+     * All rules should contain the same template name, which is returned.
+     *
+     * @throws  InvalidRulesException if the rules contains several template names
```

Comment:
Probably very nit picky but "contains" should be "contain". =P

Reply:
Thanks, I missed that!